ESP32: reset watchdog timer during update

flowerysong opened the pull request:
Without this change, updates consistently fail because the watchdog causes a panic (see https://github.com/espressif/arduino-esp32/issues/3775, where similar behaviour is reported.)

```
E (20871) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (20871) task_wdt:  - IDLE (CPU 0)
E (20871) task_wdt: Tasks currently running:
E (20871) task_wdt: CPU 0: ipc0
E (20871) task_wdt: CPU 1: loopTask
E (20871) task_wdt: Aborting.
```

With this change, OTA updates complete as expected.

ayushsharma82 commented:
Hi @flowerysong,

Thanks for digging into code and finding the cause! I'd like to merge this, can you please resolve the conflicts which occurred with recent merge? We just split AsyncElegantOTA header into separate .h and .cpp files with #95 .

flowerysong committed:
ESP32: reset watchdog timer

Fixes:

E (20871) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (20871) task_wdt:  - IDLE (CPU 0)
E (20871) task_wdt: Tasks currently running:
E (20871) task_wdt: CPU 0: ipc0
E (20871) task_wdt: CPU 1: loopTask
E (20871) task_wdt: Aborting.
